### DISCUSSION TITLE
Duplicated message can not be avoid when client reconnect ?

### DISCUSSION THREAD

**Discussion by TakaHiR07** [apache/pulsar#18631] [Q&A]
Using pulsar-2.9.3, go-client-0.6.0
I enable batch in producer, keep produce message with auto-increment Id. Then Consumer receive message, write message to other database, individualAck message, this process is sync, make sure the message is write into database and ack success. There has no duplicated message, it's right. But when I execute consumer.close(), then restart it, there would occur several duplicated message with same Id at this point of time. 

I have try setting config acknowledgmentAtBatchIndexLevelEnabled=true, duplicated message still occur. It seem the acked message has been redeliver to consumer when reconnecting.

Is duplicated message reasonable when client reconnect, although the message has been acked ?

**Comment by chamons** [comment]
I believe I'm seeing that as well, in our case we are restarting a consumer pool adding and removing one node at a time once it passes health checks.
